server/raft.go

```diff
@@ -3680,11 +3674,11 @@ func (n *raft) writePeerState(ps *peerState) {
 	if bytes.Equal(n.wps, pse) {
 		return
 	}
-	// Stamp latest and kick writer.
+	// Stamp latest and write the peer state file.
 	n.wps = pse
-	select {
-	case n.wpsch <- struct{}{}:
-	default:
+	if err := writePeerState(n.sd, ps); err != nil {
+		n.setWriteErr(err)
```

Comment:
Should we check that we are not closed like we do in the async version or is that handled above?

Reply:
Ah yes, have added it back in.